lib/singlepass-backend/src/codegen_x64.rs

```diff
@@ -1156,7 +1158,7 @@ impl ModuleCodeGenerator<X64FunctionCode, X64ExecutionContext, CodegenError>
     unsafe fn from_cache(artifact: Artifact, _: Token) -> Result<ModuleInner, CacheError> {
         let (info, _, memory) = artifact.consume();
 
-        let cache_image: CacheImage = bincode::deserialize(memory.as_slice())
+        let cache_image: CacheImage = BorshDeserialize::deserialize(&mut memory.as_slice())
```

Comment:
Hm, shouldn't this be `BorshDeserialize::try_from_slice`? Ie, seems like at the moment we don't check for leftover data?

Reply:
I remembered i used try_from_slice and failed, because at serialization time, wasmer reuse the buffer for additional thing, so at deserization it's _expected to not ended_. And bincode::deserialize is a non-end-checking version too. But it's long time ago let me double check.

Reply:
Yup, you are correct, `bincode::deserialize` allows trailing bytes.

Reply:
Ah try_from_slice actually works and there should be no leftover data by proof read context source code + rerun. what happened here:
1. I had a wrong borsh deserialize implementation on some sub-struct, and as a result it shown as fail of there's leftover data, so I first attempt to replace try_from_slice with deserialize: https://github.com/near/wasmer/pull/1/commits/aa0daae4abbb99eb6fca1e8d7b803d009300b331
2. I then added more debug prints and fixed the deserialization implementation. it is fixed since 5a1dac96e5691c7dc71679d151aecaab0e75dc38 and since then, it also should be no leftover data, but I completely forgot of changing it back to try_from_slice

Reply:
Hm, interesting, bincode allows trailing bytes nontheless: https://docs.rs/bincode/1.3.3/src/bincode/lib.rs.html#177-185. 

I guess this needs another careful look at, will do after the meeting!